src/test/java/hudson/tasks/junit/JUnitResultArchiverTest.java

```diff
@@ -330,4 +338,40 @@ public String getName() {
         wc.waitForBackgroundJavaScript(10000L);
         assertThat(page.asText(), not(containsString(EXPECTED)));
     }
+
+    @Issue("JENKINS-26535")
+    @Test
+    public void testDescribableRoundTrip() throws Exception {
+        DescribableModel<JUnitResultArchiver> model = new DescribableModel<JUnitResultArchiver>(JUnitResultArchiver.class);
+        Map<String,Object> args = new TreeMap<String,Object>();
+
+        args.put("testResults", "**/TEST-*.xml");
+        JUnitResultArchiver j = model.instantiate(args);
+        assertEquals("**/TEST-*.xml", j.getTestResults());
+        assertFalse(j.isAllowEmptyResults());
+        assertFalse(j.isKeepLongStdio());
+        assertEquals(1.0, j.getHealthScaleFactor(), 0);
+        assertTrue(j.getTestDataPublishers().isEmpty());
+        assertEquals(args, model.uninstantiate(model.instantiate(args)));
+
+        // Test roundtripping of an instantiated list of test data publishers.
+        args.put("testDataPublishers", Collections.<TestDataPublisher>singletonList(new MockTestDataPublisher("test")));
+        Map<String,Object> explicit = model.uninstantiate(model.instantiate(args));
+        JUnitResultArchiver j2 = model.instantiate(explicit);
+        assertFalse(j2.getTestDataPublishers().isEmpty());
+
+        assertEquals(explicit, model.uninstantiate(model.instantiate(explicit)));
+
+        // Test roundtripping from a Pipeline-style describing of the publisher.
+        Map<String,String> describedPublisher = new HashMap<String, String>();
+        describedPublisher.put("$class", "MockTestDataPublisher");
+        describedPublisher.put("name", "test");
+        args.put("testDataPublishers", Collections.singletonList(describedPublisher));
+
+        Map<String,Object> described = model.uninstantiate(model.instantiate(args));
+        JUnitResultArchiver j3 = model.instantiate(described);
+        assertFalse(j3.getTestDataPublishers().isEmpty());
```

Comment:
Best to go ahead and verify that the list has one element, which is a `MockTestDataPublisher`, with the right `getName`.